### PR TITLE
make it possible to add customer request/response interceptors to sel…

### PR DESCRIPTION
**Pull request by asolntsev**
…enide proxy server

## Proposed changes
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java [hunk 1]
```diff
@@ -67,12 +67,12 @@ public void start() {
     port = proxy.getPort();
   }
 
-  private void addRequestFilter(String name, RequestFilter requestFilter) {
+  public void addRequestFilter(String name, RequestFilter requestFilter) {
```
Comment (rosolko) [inline src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java]: `private` to `public` also required javadoc.
So checkstyle failed.

File: src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java [hunk 2]
```diff
@@ -67,12 +67,12 @@ public void start() {
...
     proxy.addRequestFilter(requestFilter);
     requestFilters.put(name, requestFilter);
   }
 
-  private void addResponseFilter(String name, ResponseFilter responseFilter) {
+  public void addResponseFilter(String name, ResponseFilter responseFilter) {
```
Comment (rosolko) [inline src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java]: Same here (javadoc).

Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java]: fixed

File: src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java (hunk 1 above)
Comment (asolntsev) [reply src/main/java/com/codeborne/selenide/proxy/SelenideProxyServer.java]: fixed

**Pull request merged**